packages/flutter/lib/src/foundation/persistent_hash_map.dart

```diff
@@ -0,0 +1,388 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// A collection of key/value pairs, from which you can retrieve a value
+/// using its associated key.
+///
+/// This class implements a persistent map: extending this map with a new
+/// key/value pair does not modify an existing instance but instead creates a
+/// new instance.
+///
+/// Note: unlike [Map] this class does not support `null` as a key value and
+/// implements only a functionality needed for a specific use case at the
+/// core of the framework.
```

Comment:
No, I would really like to avoid that, but I could not figure out how. It does not seem to be possible. Any suggestions? 

Originally I just wanted to reference it as `package:flutter/src/foundation/persistent_hash_map.dart` making it clear that it is not a public API - but there is a lint check that prevents this sort of stuff. 

Alternative is to put it in `widgets/` and use just there - but it felt misplaced there (and I am not sure that tests would be able to import `package:flutter/src/widgets/persistent_hash_map.dart`

Any idea on how to approach this?